crates/telio-batcher/src/batcher.rs

```diff
@@ -104,6 +168,41 @@ where
             interval,
             threshold,
         );
+
+        let interval = {
+            if interval < MIN_INTERVAL {
+                telio_log_warn!(
+                    "Interval of {:?} is too short, overriding to {:?}",
+                    interval,
+                    MIN_INTERVAL
+                );
+                MIN_INTERVAL
+            } else {
+                interval
+            }
+        };
+
+        let max_threshold = {
+            if interval.as_secs() % 2 == 0 {
+                interval / 2
+            } else {
+                interval / 2 + Duration::from_secs(1)
```

Comment:
so if min intreval is 5 and max threshold becomes 3, will i not casue any issues in scenarion like:
```
A: --**d--**d--**d
B: *d--**d--**d--
```
Basicaly both A and B can ajust, and it's uclear witch one will do this, but maybe this is ok ?

Reply:
disclaimer: I think it would be safer to have just `/ 2` without the +1 but I see no harm in it except that it ensures that it should eventually batch.

Given your example:
```
A: --**D--**d--**d
B: *d--**d--**d--
```
Actually it will batch on A's deadline since B's threshold falls right on it